### PR TITLE
`:external-type` option now takes the TS type name

### PR DESCRIPTION
**Pull request by tiagodalloca**
The right way of using external types:

```clojure
(register! ::inst [inst? {::mts/external-type "Date" ::mts/t-name "Inst"}])
```

This will give you

```ts
type Inst = Date
```

There are changes in `mts/external-type` function signature:
```clojure
(defn external-type
  [external-type-name & {:keys [import-path import-alias type-name schema]}] ,,,)
```

Plus minor fix that was documented as a new test

### CODE REVIEW

File: src/malli_ts/core.cljc [hunk 1]
```diff
@@ -37,55 +50,55 @@
     :else [:type :any]))
 
 (defmulti ^:private -parse-ast-node
-  #'-dispatch-parse-ast-node)
```
Comment (flow-danny) [inline src/malli_ts/core.cljc]: What happened to whitespace?

File: src/malli_ts/core.cljc [hunk 2]
```diff
@@ -37,55 +50,55 @@
...
-                               (if (= (take-last 2 $) ["d" "ts"])
-                                 (drop-last 2 $)
-                                 $)
-                               (string/join "_" $)))
-                            ref-file))
-          ref-type-name (get-in schema-id->type-options [$ref :t-name])
-          same-file? (= file ref-file)]
-      (when-not same-file?
-        (swap! file-imports* update file set/union #{ref-file}))
-      (str (if-not same-file? (str import-alias ".") nil) ref-type-name))))
```
Comment (flow-danny) [inline src/malli_ts/core.cljc]: Nothing changed but 2 spaces indent?

File: src/malli_ts/core.cljc [hunk 3]
```diff
@@ -160,7 +177,7 @@
       (swap! file-imports* update file set/union #{t-path}))
     (when (and import-alias (not canonical-alias))
       (swap! files-import-alias* assoc t-path-str import-alias))
-    (str (if import-alias (str import-alias ".") nil) t-name)))
+    (str (if import-alias (str import-alias ".") nil) external-type)))
```
Comment (flow-danny) [inline src/malli_ts/core.cljc:180]: Is `external-type` validated to never be empty string here? Maybe have a fallback to the schema name?

File: test/malli_ts/core_test.cljc [hunk 4]
```diff
@@ -73,34 +73,43 @@
...
     
     ::answerUltimateQuestion [:=> :cat :random.dir.universe/answer-to-everything]
 
-    :global/date (mts/external-type "Date")
+    :global/date (mts/external-type "Date" :schema inst?)
     ::now [:=> [:cat] :global/date]
 
     ;; explicit "absolute" path (already default )
-    :crypto/hash (mts/external-type "Hash" {:absolute "crypto"} "crypto")
+    :crypto/hash (mts/external-type "Hash" :import-path {:absolute "crypto"} :import-alias "crypto")
```
Comment (flow-danny) [inline test/malli_ts/core_test.cljc:83]: Nice, now you can update README.md 😄

**Review by flow-danny — CHANGES_REQUESTED** [review]

File: src/malli_ts/core.cljc (hunk 1 above)
Comment (tiagodalloca) [reply src/malli_ts/core.cljc]: fixed

File: src/malli_ts/core.cljc (hunk 2 above)
Comment (tiagodalloca) [reply src/malli_ts/core.cljc]: fixed

File: src/malli_ts/core.cljc (hunk 3 above)
Comment (tiagodalloca) [reply src/malli_ts/core.cljc:180]: Being an `external-type`, I don't think falling back to schema's name makes sense.

Not only that, `-dispatch-parse-ast-node` will only dispatch to `:external-type` if `::mts/external-type` is not `nil`

Comment (tiagodalloca) [reply src/malli_ts/core.cljc:180]: Guaranteed `:external-type` is not `nil`

File: test/malli_ts/core_test.cljc (hunk 4 above)
Comment (tiagodalloca) [reply test/malli_ts/core_test.cljc:83]: Done!

File: src/malli_ts/core.cljc (hunk 3 above)
Comment (flow-danny) [reply src/malli_ts/core.cljc:180]: Ah I missed that context, thanks!

**Review by flow-danny — APPROVED** [review]

**Pull request merged**